app/src/main/java/com/radixdlt/android/ui/activity/ReceiveRadixInvoiceActivity.kt

```diff
@@ -37,51 +52,137 @@ class ReceiveRadixInvoiceActivity : BaseActivity() {
     }
 
     override fun onCreate(savedInstanceState: Bundle?) {
-//        AndroidInjection.inject(this)
+        AndroidInjection.inject(this)
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_receive_radix_invoice)
 
         showAddress()
 
-        val qrCode = QRCode.from(myAddress)
-            .withSize(size, size)
-            .withHint(EncodeHintType.MARGIN, 1)
-            .bitmap()
-
-        Glide.with(this).load(qrCode).into(imageView)
-
         setSupportActionBar(toolbar as Toolbar)
         supportActionBar?.title = getString(R.string.receive_radix_activity_title)
         supportActionBar?.setDisplayHomeAsUpEnabled(true)
 
         setListeners()
-//        initialiseViewModels()
+        initialiseViewModels()
+    }
+
+    private fun generateURI(
+        address: String = myAddress,
+        token: String = GENESIS_XRD,
+        amount: String = "O",
+        attachment: String? = null
+    ): String {
+        val domain = "https://www.radixdlt.com"
+        val schema = "/dapp/payment/send"
```

Comment:
Nitpicking, but to be perfectly correct the `schema` should actually be called `path` given the value it holds. I realise this most likely is a refactoring artefact, but still... :)

Reply:
Yea I did not think about the names at all if I am honest. I will change it so I don't forget!